catalog/async/catalog-async-processingframework/src/main/java/org/codice/ddf/catalog/async/processingframework/impl/InMemoryProcessingFramework.java

```diff
@@ -0,0 +1,257 @@
+            LOGGER.debug("postProcessPlugins is empty. Not starting post process thread");
+        } else {
+            threadPool.submit(() -> {
+                ProcessRequest<ProcessCreateItem> request = input;
+
+                for (PostProcessPlugin plugin : postProcessPlugins) {
+                    try {
+                        request = plugin.processCreate(request);
+                    } catch (PluginExecutionException e) {
+                        LOGGER.debug("Unable to process create request through plugin: {}",
+                                plugin.getClass()
+                                        .getCanonicalName(),
+                                e);
+                    }
+                }
+
+                storeProcessRequest(request);
+            });
+        }
+    }
+
+    @Override
+    public void submitUpdate(ProcessRequest<ProcessUpdateItem> input) {
+        if (postProcessPlugins == null || postProcessPlugins.isEmpty()) {
+            LOGGER.debug("postProcessPlugins is empty. Not starting post process thread");
+        } else {
+            threadPool.submit(() -> {
+                ProcessRequest<ProcessUpdateItem> request = input;
+
+                for (PostProcessPlugin plugin : postProcessPlugins) {
+                    try {
+                        request = plugin.processUpdate(request);
+                    } catch (PluginExecutionException e) {
+                        LOGGER.debug("Unable to process update request through plugin: {}",
+                                plugin.getClass()
+                                        .getCanonicalName(),
+                                e);
+                    }
+                }
+
+                storeProcessRequest(request);
+            });
+        }
+    }
+
+    @Override
+    public void submitDelete(ProcessRequest<ProcessDeleteItem> input) {
+        if (postProcessPlugins == null || postProcessPlugins.isEmpty()) {
+            LOGGER.debug("postProcessPlugins is empty. Not starting post process thread");
+        } else {
+            threadPool.submit(() -> {
+                ProcessRequest<ProcessDeleteItem> request = input;
+
+                for (PostProcessPlugin plugin : postProcessPlugins) {
+                    try {
+                        request = plugin.processDelete(request);
+                    } catch (PluginExecutionException e) {
+                        LOGGER.debug("Unable to process request through plugin: {}",
+                                plugin.getClass()
+                                        .getCanonicalName(),
+                                e);
+                    }
+                }
+            });
+        }
+    }
+
+    private <T extends ProcessResourceItem> void storeProcessRequest(
+            ProcessRequest<T> processRequest) {
+        LOGGER.trace("Storing update request post processing change(s)");
+
+        Set<ContentItem> contentItemsToUpdate = new HashSet<>();
```

Comment:
Yes that is the reason. We did not want to send back wasted duplicate updates in case a Plugin decided to do something weird. The Object `hashCode()` would not be enough in this case?